Feature search

Yama-aa opened the pull request:
Added SearchMovie component - the View has a text box and a search button that sends an http request.
Added to the server a new api for movie search that returns a default JSON.
The results are shown in the same page, with a link to the movie page.
Implemented redux into the feature.

Commit:
search:
Added SearchMovieView component with a text box and a search button that sends an http request.
Added to the server a new api for movie search that returns a default JSON.
The results are shown in the same page, with a link to the movie page.

Commit:
implemented redux into search feature

Merge branch 'master' of https://github.com/llmaayanll/rate-movies into feature-search

# Conflicts:
#	package-lock.json
#	package.json

yarindeoh reviewed (changes requested):
Great start, well done!!! :)
1. you can add 'services' to webpack.config alias, so you wont use relative paths (i.e '../../) :
services: path.resolve(__dirname, 'frontend/src/services'),
2. you didn't used the consts you've created in searchMovieConstants.js. Use it in the reducer and every time you want to use that consts. (by import SET_QUERY from '***')

- next step (not is this PR) is to create a a db query to get all movies that contain the search query

Commit:
acoording to code review:
1) added 'services' to webpack.config alias and updated SearchMovieView
2) updated the reducer to use searchMovieConstants
